onnxruntime/core/framework/execution_frame.cc

```diff
@@ -45,6 +45,16 @@ OrtValue* IExecutionFrame::GetMutableNodeInputOrOutputMLValue(int index) {
   return const_cast<OrtValue*>(GetNodeInputOrOutputMLValue(index));
 }
 
+Status IExecutionFrame::SetOutputMLValue(int index, const OrtValue& ort_value) {
```

Comment:
>SetOutputMLValue [](start = 24, length = 16)

could this function be used to set inputs too?
IExecutionFrame::GetNodeInputOrOutputMLValue() above calls the same GetNodeIdxToMLValueIdx() function

Reply:
I don't think so... 
each index slot should still be assigned only once. This SetOutputMLValue is overriding the original **unassigned** index slot. 

If we allow this on the inputs, a slot can assigned more than once: first assigned by some node's output, and then assigned by "SetInputMLValue". Such behavior is undesirable.